### PR TITLE
Raise an Err when CRYPTO_THREAD_lock_new fails

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Add missing error raise call, as it is done everyxhere else.

### CODE REVIEW

File: crypto/asn1/tasn_utl.c [hunk 1]
```diff
@@ -117,8 +117,10 @@ int asn1_do_lock(ASN1_VALUE **pval, int op, const ASN1_ITEM *it)
     if (op == 0) {
         *lck = 1;
         *lock = CRYPTO_THREAD_lock_new();
-        if (*lock == NULL)
+        if (*lock == NULL) {
+            ASN1err(ASN1_F_ASN1_DO_LOCK, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/asn1/tasn_utl.c]: For this function, that's inconsistent behavior.  There are other `return 0` with no error code being set.

File: crypto/x509/by_dir.c [hunk 2]
```diff
@@ -152,12 +152,14 @@ static int new_dir(X509_LOOKUP *lu)
     if ((a = OPENSSL_malloc(sizeof(*a))) == NULL)
         return 0;
     if ((a->buffer = BUF_MEM_new()) == NULL) {
+        X509err(X509_F_NEW_DIR, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/x509/by_dir.c]: No.  Inconsistent behavior

File: crypto/x509/by_dir.c [hunk 3]
```diff
@@ -152,12 +152,14 @@ static int new_dir(X509_LOOKUP *lu)
...
         OPENSSL_free(a);
         return 0;
     }
     a->dirs = NULL;
     a->lock = CRYPTO_THREAD_lock_new();
     if (a->lock == NULL) {
+        X509err(X509_F_NEW_DIR, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/x509/by_dir.c]: No.  Inconsistent behavior

File: crypto/asn1/tasn_utl.c (hunk 1 above)
Comment (richsalz) [inline crypto/asn1/tasn_utl.c]: On the other hand, we try to set malloc-failure every time malloc fails. Software is a work in progress :)

File: include/openssl/x509.h [hunk 4]
```diff
@@ -1060,6 +1060,7 @@ void ERR_load_X509_strings(void);
 # define X509_F_GET_CERT_BY_SUBJECT                       103
 # define X509_F_NETSCAPE_SPKI_B64_DECODE                  129
 # define X509_F_NETSCAPE_SPKI_B64_ENCODE                  130
+# define X509_F_NEW_DIR                                   150
```
Comment (levitte) [inline include/openssl/x509.h]: Remove

File: include/openssl/asn1.h [hunk 5]
```diff
@@ -929,6 +929,7 @@ void ERR_load_ASN1_strings(void);
 # define ASN1_F_ASN1_D2I_READ_BIO                         107
 # define ASN1_F_ASN1_DIGEST                               184
 # define ASN1_F_ASN1_DO_ADB                               110
+# define ASN1_F_ASN1_DO_LOCK                              233
```
Comment (levitte) [inline include/openssl/asn1.h]: Remove

**Comment by levitte** [comment]
So, some of them no, others yes.  Please adjust?

File: crypto/asn1/tasn_utl.c (hunk 1 above)
Comment (levitte) [inline crypto/asn1/tasn_utl.c]: I'm still going to argue that an individual function should work *consistently*, so for this one, there should be an error code for *each* `return 0` or none at all.

Comment (FdaSilvaYY) [inline crypto/asn1/tasn_utl.c]: OK, any error in this method is not handled by his callers.

File: crypto/x509/by_dir.c (hunk 2 above)
Comment (FdaSilvaYY) [inline crypto/x509/by_dir.c]: My mistake, Err is already set inside BUF_MEM_new.

File: crypto/x509/by_dir.c (hunk 3 above)
Comment (FdaSilvaYY) [inline crypto/x509/by_dir.c]: I don't see why exactly ? I commented it.

**Comment by levitte** [comment]
Looking good.  :+1: pending build results.

File: crypto/asn1/tasn_utl.c [hunk 6]
```diff
@@ -117,8 +117,10 @@ int asn1_do_lock(ASN1_VALUE **pval, int op, const ASN1_ITEM *it)
     if (op == 0) {
         *lck = 1;
         *lock = CRYPTO_THREAD_lock_new();
-        if (*lock == NULL)
+        if (*lock == NULL) {
+            //ASN1err(ASN1_F_ASN1_DO_LOCK, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/asn1/tasn_utl.c]: C comment style, please

File: crypto/x509/by_dir.c [hunk 7]
```diff
@@ -158,6 +158,7 @@ static int new_dir(X509_LOOKUP *lu)
     a->dirs = NULL;
     a->lock = CRYPTO_THREAD_lock_new();
     if (a->lock == NULL) {
+        //X509err(X509_F_NEW_DIR, ERR_R_MALLOC_FAILURE);
```
Comment (levitte) [inline crypto/x509/by_dir.c]: C comment style, please

**Comment by levitte** [comment]
:+1:

**Comment by mattcaswell** [comment]
Why are we adding *new* commented out code? That looks odd.

**Comment by FdaSilvaYY** [comment]
asn1_do_lock method does not return any error value,
as (not) stated in the block comment just above it.
So it's pointless to raise an error here...  so i keep it as a comment .

**Comment by mattcaswell** [comment]
I don't really think the comment is adding any value here. I'd rather not add commented out code.

**Comment by FdaSilvaYY** [comment]
It should be a TODO: return an error code, and let the caller manage it ... 
No ?

**Comment by mattcaswell** [comment]
Well either just fix the problem so that it does return an error code (this is an internal function so we can change it) - or if we must then, yes - add a TODO comment.

**Comment by FdaSilvaYY** [comment]
Hi,  I add a FIXME on this line.
The patch to clear this point is on my stack, i will push it after.

**Comment by mattcaswell** [comment]
You still left the commented out code in this patch. That should be removed.

**Comment by mattcaswell** [comment]
+1

@levitte can you reconfirm your +1?

**Comment by mattcaswell** [comment]
Ping @levitte

**Comment by levitte** [comment]
Reconfirming +1

**Comment by mattcaswell** [comment]
Merged as b2b361f6a. Thanks.

**Pull request closed without merging**